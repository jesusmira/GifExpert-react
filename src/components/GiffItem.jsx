import PropTypes from 'prop-types';


export const GiffItem = ({title, url, id}) => {
  return (
    <div className="card">
        <img src={ url } alt={ title } />
        <p>{ title }</p>
    </div>

  )
}

GiffItem.propTypes = {

    title: PropTypes.string.isRequired,
    url  : PropTypes.string.isRequired

  }

/* Tarea:

  1.- AÃ±adir PropTypes.
    a title obligatorio
    b url obligatorio

  2. Evaluar snapshot.
*/ 


