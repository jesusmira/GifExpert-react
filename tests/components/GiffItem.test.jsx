import  { render, screen } from '@testing-library/react';
import  { GiffItem } from '../../src/components/GiffItem';


describe ('Pruebas de GifItem', () => { 

   const title = 'Saitama';
   const url = 'https://one-punch.com/saitama.jpg';

   test('Debe de hacer match con el snaphot', () => { 
      
      const { container } = render( <GiffItem title={ title } url = { url}  />)
      expect( container ).toMatchSnapshot();
   });

   test('Debe de mostrar la imagen con el URL y el Alt indicado', () => { 
      render( <GiffItem title={ title } url = { url}  />);
      // screen.debug();
      // expect ( screen.getByRole("img").src).toBe( url );
      // expect ( screen.getByRole("img").alt).toBe( title );
      const {src, alt} = screen.getByRole("img");
      expect (src).toBe(url);
      expect (alt).toBe(alt);

   });

   test('Debe de mostrar el titulo en el componente', () => { 
      render( <GiffItem title={ title } url = { url}  />);
      expect(screen.getByText( title )).toBeTruthy();
   });
   
});