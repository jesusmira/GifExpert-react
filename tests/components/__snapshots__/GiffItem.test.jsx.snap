// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifItem Debe de hacer match con el snaphot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
